docs/source/advanced.rst

```diff
@@ -258,43 +258,187 @@ of :py:class:`scikeras.wrappers.BaseWrappers`. Currently, they are:
 
 All routed parameters will be available for hyperparameter tuning.
 
+For example:
+
+.. code:: python
+
+   clf = KerasClassifier(..., fit__batch_size=32, predict__batch_size=10000)
+
 Below are some example use cases.
 
-Example: multiple losses with routed parameters
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Compilation with routed parameters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+SciKeras can compile optimizers, losses and metrics with routed parameters.
+This allows hyperparameter tuning with deeply nested parameters to optimizer, losses and metrics.
+You can use this feature both when you let SciKeras compile your Model and when you compile your own model
+within ``model_build_fn`` (by accepting the ``compile_kwargs`` parameter).
+
+Optimizer
++++++++++
 
 .. code:: python
 
-    from tensorflow.keras.losses import BinaryCrossentropy, CategoricalCrossentropy
+    from scikeras.wrappers import KerasClassifier
+    from tensorflow import keras
 
     clf = KerasClassifier(
         model=model_build_fn,
+        optimizer=keras.optimizers.SGD,
+        optimizer__learning_rate=0.05
+    )
+
+
+Losses
+++++++
+
+.. code:: python
+
+    from tensorflow.keras.losses import BinaryCrossentropy, CategoricalCrossentropy
+
+    clf = KerasClassifier(
+        ...
+        loss=BinaryCrossentropy,
+        loss__label_smoothing=0.1,  # results in BinaryCrossentropy(label_smoothing=0.1)
+    )
+    # pure Keras object only syntax is still supported, but parameters can't be tuned
+    clf = KerasClassifier(
+        ...
+        loss=BinaryCrossentropy(label_smoothing=0.1),
+    )
+
+Keras, and SciKeras by extension, support passing a single loss (which will be applied to all outputs)
+or a loss for each output by passing a list of losses or a dict of losses.
+
+Additionally, SciKeras supports routed parameters to each individual loss, or to all losses together.
+
+.. code:: python
+
+    from tensorflow.keras.losses import BinaryCrossentropy, CategoricalCrossentropy
+
+    clf = KerasClassifier(
+        ...
         loss=[BinaryCrossentropy, CategoricalCrossentropy],
         loss__from_logits=True,  # BinaryCrossentropy(from_logits=True) & CategoricalCrossentropy(from_logits=True)
         loss__label_smoothing=0.1,  # passed to each sub-item, i.e. `loss=[l(label_smoothing=0.1) for l in loss]`
```

Comment:
```suggestion
```

I'm not seeing the need for this line; it's already expressed in the line above.